exp/controllers/machinepool_controller_phases.go

```diff
@@ -232,12 +232,16 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 	// Call generic external reconciler.
 	infraReconcileResult, err := r.reconcileExternal(ctx, cluster, mp, &mp.Spec.Template.Spec.InfrastructureRef)
 	if err != nil {
-		if mp.Status.InfrastructureReady && strings.Contains(err.Error(), "could not find") {
-			// Infra object went missing after the machine pool was up and running
-			log.Error(err, "MachinePool infrastructure reference has been deleted after being ready, setting failure state")
-			mp.Status.FailureReason = capierrors.MachinePoolStatusErrorPtr(capierrors.InvalidConfigurationMachinePoolError)
-			mp.Status.FailureMessage = pointer.StringPtr(fmt.Sprintf("MachinePool infrastructure resource %v with name %q has been deleted after being ready",
-				mp.Spec.Template.Spec.InfrastructureRef.GroupVersionKind(), mp.Spec.Template.Spec.InfrastructureRef.Name))
+		if strings.Contains(err.Error(), "could not find") {
```

Comment:
```suggestion
		if apierrors.IsNotFound(errors.Cause(err)) {
```

Reply:
Thanks, I was thinking that I had seen this some place in the codebase.